src/app/notifications/notification-helper.ts

```diff
@@ -0,0 +1,74 @@
+import { ElectronNotificationData, INotificationData, NotificationActions } from '../../common/api-interface';
+import { isWindowsOS } from '../../common/env';
+import { notification } from '../../renderer/notification';
+import { windowHandler } from '../window-handler';
+import { windowExists } from '../window-utils';
+import { ElectronNotification } from './electron-notification';
+
+class NotificationHelper {
+    private electronNotification: Map<number, ElectronNotification>;
+
+    constructor() {
+        this.electronNotification = new Map<number, ElectronNotification>();
+    }
+
+    /**
+     * Displays Electron/HTML notification based on the
+     * isElectronNotification flag
+     *
+     * @param options {INotificationData}
+     */
+    public showNotification(options: INotificationData) {
+        if (options.isElectronNotification) {
+            // MacOS: Electron notification only supports static image path
+            options.icon = this.getIcon(options);
+            const electronToast = new ElectronNotification(options, this.notificationCallback);
+            this.electronNotification.set(options.id, electronToast);
+            electronToast.show();
+        } else {
+            notification.showNotification(options, this.notificationCallback);
+        }
+    }
+
+    /**
+     * Closes a specific notification by id
+     *
+     * @param id {number} - unique id assigned to a specific notification
+     */
+    public async closeNotification(id: number) {
+        if (this.electronNotification.has(id)) {
+            const electronNotification = this.electronNotification.get(id);
+            if (electronNotification) {
+                electronNotification.close();
+            }
+        } else {
+            await notification.hideNotification(id);
+        }
+    }
+
+    /**
+     * Method that sends the notification actions event
+     * on to web client
+     *
+     * @param event {NotificationActions}
+     * @param data {ElectronNotificationData}
+     * @param extraData {ElectronNotificationData}
+     */
+    public notificationCallback(
+        event: NotificationActions,
+        data: ElectronNotificationData,
+        extraData: ElectronNotificationData,
+    ) {
+        const mainWindow = windowHandler.getMainWindow();
+        if (mainWindow && windowExists(mainWindow) && mainWindow.webContents) {
+            mainWindow.webContents.send('notification-actions', { event, data, extraData });
+        }
+    }
+
+    private getIcon(options: INotificationData): string | undefined {
+        return isWindowsOS ? options.icon : undefined;
```

Comment:
What happens for macOS and Linux? Will those operating systems take app's default icon?

Reply:
Yes, That's correct